### DISCUSSION TITLE
Thanks MegaLinter team!

### DISCUSSION THREAD

**Discussion by samsatoworks** [oxsecurity/megalinter#1823] [General]
Many thanks for the dedication and hard work you have devoted to this fantastic project. Since I introduced it to my workplace in 2021, it has become an indispensable part of our code review process. We produce better, more readable, more maintainable code than ever before. I wish I would have discovered MegaLinter many years ago! You should be very proud that your efforts are helping to make better developers all over the world!

**Comment by nvuillam** [comment]
Thanks a lot @samsatoworks , that's a pleasure to receive such a nice feedback :) 

Where do you work, and on what kind of tech stack do you use MegaLinter ? ^^ ( answer the questions you can according to confidentiality ^^ )

And if you have ideas about evolution, new contributions are very welcomed :)

**Reply by FlorianPfaff** [reply]
Thank you from my side as well for creating and maintaining MegaLinter and of course thanks to everyone from OxSecurity for taking over this nice tool. I understand your question a general one (and not just directed to samsatoworks) about your userbase: I work at Karlsruhe Institute of Technology https://isas.iar.kit.edu/Staff_Pfaff.php. A lot of my research is based on Matlab (which is still popular in the tracking and data fusion community) but I am more and more transitioning to Python - enforcing adherence to good style is important to me - and megalinter makes it pretty easy (and hopefully now also efficient for me, see https://github.com/oxsecurity/megalinter/discussions/2616) to keep good style.
Concerning the tool chain: I use VS code with GitHub actions, sometimes a dedicated GIT-GUI for more complicated operations.

For me, there is no wish aobut "evolution" - it should just stay maintained with the latest linter updates and that's it! I am sure OxSecurity has a lot of other projects, and I am grateful if you just keep it alive.

**Reply by Kurt-von-Laven** [reply]
Thank you, @FlorianPfaff. We appreciate the support and are glad to hear MegaLinter is proving helpful to you! We actually use the same toolchain ourselves, although I should clarify that the we here is [ScribeMD](https://github.com/ScribeMD/) rather than Ox Security. They are MegaLinter's sponsor, and we (MegaLinter's maintainers) are very grateful for their support!

**Reply by nvuillam** [reply]
Thanks a lot for this nice feedback @FlorianPfaff , glad to see MegaLinter helps you with your work :) 

Don't hesitate to post about it on networks 😇

About @oxsecurity, they are not just sponsors ^^ See more details in the following article ^^

https://nicolas.vuillamy.fr/megalinter-sells-his-soul-and-joins-ox-security-2a91a0027628

**Reply by FlorianPfaff** [reply]
I have recommended and will keep recommending it. Though, in academia and its "this-better-be-working-before-the-paper-deadline" mentality, the interest so far is not that high 😉. But I will also make sure to recommend it to students starting new projects because they surely benefit from it, and for new projects, it is just some continual fixing here and there instead of a huge mountain of work as for existing projects.

**Reply by nvuillam** [reply]
On existing projects, you can start by defining XXX_DISABLE_ERRORS_IF_LESS_THAN, that way the CI jobs are not blocking with the current state of the project, but it prevents to add new errors ^^

Example: `PYTHON_RUFF_DISABLE_ERRORS_IF_LESS_THAN: 32`

Some linters can also be deactivated because they are considered not relevant, or meanwhile people find the time to fix issues :)

**Reply by Kurt-von-Laven** [reply]
We experienced the same pain point when we first started using MegaLinter and created the [`megalinter-incremental`](.pre-commit-hooks.yaml) [pre-commit](https://pre-commit.com) hook for this purpose, which only checks modified files. Some linters only run on the entire project, so if you want to run those as well, the [`megalinter-full`](.pre-commit-hooks.yaml) hook serves that purpose.
